src/Features/Core/Portable/FileHeaders/AbstractFileHeaderHelper.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Formatting;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FileHeaders
+{
+    internal abstract class AbstractFileHeaderHelper
+    {
+        internal abstract int SingleLineCommentTriviaKind { get; }
+        internal abstract int MultiLineCommentTriviaKind { get; }
+        internal abstract int WhitespaceTriviaKind { get; }
+        internal abstract int EndOfLineTriviaKind { get; }
+        internal abstract string CommentPrefix { get; }
+
+        protected abstract string? TryGetTextContextOfComment(SyntaxTrivia commentTrivia);
+
+        internal FileHeader ParseFileHeader(SyntaxNode root)
```

Comment:
don't love this either.  would like to know if we can reuse our existing logic for 'file banners'.

Reply:
➡️ https://github.com/dotnet/roslyn/issues/42203